src/components/navBar/navBar.js

```diff
@@ -134,47 +178,183 @@ function MdNavBar($mdAria) {
  * @param {!angular.Scope} $scope
  * @param {!angular.Timeout} $timeout
  * @param {!Object} $mdConstant
+ * @param {!Object} $mdUtil
  * @constructor
  * @final
  * @ngInject
  */
-function MdNavBarController($element, $scope, $timeout, $mdConstant) {
-  // Injected variables
-  /** @private @const {!angular.Timeout} */
-  this._$timeout = $timeout;
+function MdNavBarController($element, $scope, $timeout, $mdConstant, $mdUtil) {
+    // Injected variables
+    /** @private @const {!angular.Timeout} */
+    this._$timeout = $timeout;
+
+    /** @private @const {!angular.Scope} */
+    this._$scope = $scope;
+
+    /** @private @const {!Object} */
+    this._$mdConstant = $mdConstant;
+
+    // Data-bound variables.
+    /** @type {string} */
+    this.mdSelectedNavItem;
+
+    /** @type {string} */
+    this.navBarAriaLabel;
+
+    // State variables.
+
+    /** @type {?angular.JQLite} */
+    this._navBarEl = $element[0];
+
+    /** @type {?angular.JQLite} */
+    this._inkbar;
+
+    var self = this;
+    // need to wait for transcluded content to be available
+    var deregisterTabWatch = this._$scope.$watch(function () {
+            return self._navBarEl.querySelectorAll('._md-nav-button').length;
+        },
+        function (newLength) {
+            if (newLength > 0) {
+                self._initTabs();
+                deregisterTabWatch();
+            }
+        });
+
+    // Responsivity actions
+
+    /** @type {Object} */
+    this.updatePagination = $mdUtil.debounce(updatePagination, 100);
+
+    /** @type {boolean} */
+    this.shouldPaginate = shouldPaginate();
+
+    /** @type {Function} */
+    this.nextPage = nextPage;
+
+    /** @type {Function} */
+    this.previousPage = previousPage;
 
-  /** @private @const {!angular.Scope} */
-  this._$scope = $scope;
+    /** @type {Function} */
+    this.canPageBack = canPageBack;
 
-  /** @private @const {!Object} */
-  this._$mdConstant = $mdConstant;
+    /** @type {Function} */
+    this.canPageForward = canPageForward;
```

Comment:
These sound like booleans, not at least a function that returns a Boolean.

If it's the latter, I'd change the type to function(): boolean to be explicit.